README.md

```diff
@@ -1,5 +1,101 @@
 # AE Data Flow
 
+> ![CAUTION]
+> This repo is new and very WIP.
+
 This is the primary repository for the data pipelines of Application Engineering team at the NYC Department of City Planning (DCP).
 
 It is used to populate databases used by APIs.
+
+## Local setup
+
+### Make sure you're using the correct version of Node
+
+The `.nvmrc` file tells you which version of node you should be using to run the project.
+
+If you're using [nvm](https://github.com/nvm-sh/nvm) and already have the correct version installed, you can switch by running `nvm use` from the root of this repo.
+
+### Make sure you're using the correct version of python
+
+The `.python-version` file tells you which version of python you should be using to run the project.
+
+If you're using [pyenv](https://github.com/pyenv/pyenv), you can create a virtual environment by running `pyenv virtualenv venv_ae_data_flow` and switch to it by using `pyenv activate venv_ae_data_flow`.
+
+If you need to install pyenv, follow [these instructions](https://github.com/pyenv/pyenv?tab=readme-ov-file#installation).
+
+### Install dependencies
+
+Once you have cloned this repo, install the necessary dependencies:
+
+```bash
+npm i
+pip install --editable .
+```
+
+### Set environment variables
+
+Create a file called `.env` in the root folder of the project and copy the contents of `sample.env` into that new file.
+
+Next, fill in the blank values and edit `BUILD_ENGINE_SCHEMA=local_YOUR_NAME` to be unique amongst the team.
+
+> [!IMPORTANT]
+> To use a local database, `sample_local.env` likely has the environment variable values you need.
+>
+> To use a deployed database in Digital Ocean, the values you need can be found in the AE 1password vault.
+
+To set those environment variables in a terminal window run `export $(cat .env | sed 's/#.*//g' | xargs)`.
+
+### Test database connection
+
+```bash
+dbt debug
+```
+
+## Running the project locally
+
+> [!NOTE]
+> The approaches to running this locally will inform how this will be run in CI using github actions.
+
+### Load source data
+
+To copy source data files from Digital Ocean:
+
+```bash
+# for public files
+curl --remote-name https://BUCKET.URL.com/folder/folder/data.csv
+
+# for non-public files
+# requires MinIO
+curl -O https://dl.min.io/client/mc/release/linux-amd64/mc \
+    && chmod +x mc \
+    && mv ./mc /usr/bin/
+
+mc config host add spaces $DO_SPACES_ENDPOINT $DO_SPACES_ACCESS_KEY_ID $DO_SPACES_SECRET_ACCESS_KEY --api S3v4
+
+mc cp spaces/${DO_SPACES_BUCKET_DISTRIBUTIONS}/dcp_pluto/23v3/pluto.csv pluto.csv
+mc cp spaces/${DO_SPACES_BUCKET_DISTRIBUTIONS}/dcp_pluto/23v3/attachments/zoning_districts.csv zoning_districts.csv
+mc cp spaces/${DO_SPACES_BUCKET_DISTRIBUTIONS}/dcp_pluto/23v3/attachments/source_data_versions.csv source_data_versions.csv
```

Comment:
I realized that the original implementation of all this relied on files being publicly accessible and so no credentials were needed. That's why approaches like `ADD  --chown=postgres $STORAGE_URL/tables/tax_lot.csv tax_lot.csv` in `ae-zoning-api/db/Dockerfile` [here](https://github.com/NYCPlanning/ae-zoning-api/blob/main/db/Dockerfile#L9) worked.

I remember this being a compromise to experiment quickly for AE's PoC App, but DE definitely doesn't plan on setting `edm-distributions` to be accessible to anyone with a link.

We use MinIO extensively in DE to handle simple file movement to/from our DO buckets.